doc/For developers/Developer guide/Performance/Testing.md

```diff
@@ -0,0 +1,342 @@
+| Retransmits         | 257      | 335        | 2333                 | TBD                    | 1                   | 1034                            |
+| Errors              | 0        | 0          | 0                    | TBD                    | 0                   | 0                               |
+| Avg. CPU %          | 14.799   | 15.843     | 57.098               | TBD                    | 29.373              | 34.544                          |
+
+On `receiver` nodes:
+
+| Metric              | Raw      | Containers | Colocated Containers | Containers w/ Contrail | Containers (no seg) | Containers w/ Contrail (no seg) |
+|---------------------|----------|------------|----------------------|------------------------|---------------------|---------------------------------|
+| Throughput (Mbit/s) | 7375.198 | 1592.698   | 1948.824             | TBD                    | 235.784             | 98.732                          |
+| Retransmits         | 0        | 0          | 2353                 | TBD                    | 0                   | 0                               |
+| Errors              | 0        | 0          | 0                    | TBD                    | 0                   | 0                               |
+| Avg. CPU %          | 21.161   | 42.278     | 57.086               | TBD                    | 25.064              | 25.763                          |
+
+## Conclusions
+
+Conclusions:
+
+- Enabling Hyper-V on Windows Server 2016 reduces TCP throughput by a factor of 3-4.
+    - Reduced throughput can be explained by lack of support for VMQ in vmxnet3 adapters.
+- Difference between TCP throughput scenario where containers are colocated and scenario where containers are on separate nodes, suggests that VMSwitch is a bottleneck.
+- Comparing `Containers (no seg)` test with `Containers w/ Contrail (no seg)` shows that vRouter code paths could account for 50-60% drop in TCP throughput.
+
+Performance baseline for Contrail Windows should be based on network performance of containers running on Hyper-V.
+
+
+## Test environment assumptions
+
+Performance is evaluated on 2 VMware virtual machines.
+Virtual machines should have the following specs:
+
+- VMware Virtual Machine version 13,
+- CPU: 2 vCPU,
+- RAM: 4 GB,
+- Storage: 50 GB HDD (VMware Paravirtual SCSI),
+- NIC1: 1Gb (Emulated E1000E),
+- NIC2: 10Gb (VMXNET3).
+
+Both virtual machines should be spawned on the same ESXi.
+NIC1 is used as a MANAGEMENT plane adapter and can be attached to any vSwitch.
+NIC2 is used as a CONTROL and DATA adapter and it should be attached to vSS created solely
+for performance testing.
+This vSS should not have any physical NICs attached.
+
+
+## Test scenarios
+
+### Raw
+
+Description:
+
+- 2 Windows Server compute nodes (node A and node B);
+- compute nodes are configured without Hyper-V and Containers;
+- node A and node B exchange TCP segments using `NTttcp` tool;
+- used `NTttcp` options:
```

Comment:
Not sure why, but in the preview, there is a lot of space between these points...